File: 75. Sort Colors/sortColors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */

function sortColors(nums) {
    let left = 0;
    let right = nums.length - 1;

    let idx = 0;


    while (idx <= right) {
        if (nums[idx] === 0) {
            //SWIP WITH LEFT
            [nums[left], nums[idx]] = [nums[idx], nums[left]]
            left++;
            idx++;
        }
        else if (nums[idx] === 2) {
            //SWIP WITH RIGHT
            [nums[right], nums[idx]] = [nums[idx], nums[right]]
            right--;
        }
        else {
            idx++;
        }
    }
}

File: 75. Sort Colors/sortColors.cpp
#include <vector>
#include<algorithm>
#include <iostream>

using namespace std;

class Solution {
public:
    void sortColors(vector<int>& nums) {
        int left=0;
        int idx = 0;

        int right = nums.size() - 1;


        while(idx <= right){
            if(nums[idx] == 0){
                //SWIP WITH LEFT
                swap(nums[left], nums[idx]);
                left++;
                idx++;
            }
            else if(nums[idx] == 2){
                //SWIP WITH RIGHT
                swap(nums[right] , nums[idx]);
                right--;
            }
            else{
                idx++;
            }
        }
    }
};